execution/executor/src/components/chunk_output.rs

```diff
@@ -187,12 +250,21 @@ impl ChunkOutput {
         state_view: Arc<CachedStateView>,
         maybe_block_gas_limit: Option<u64>,
     ) -> Result<Vec<TransactionOutput>> {
-        Ok(V::execute_block_sharded(
-            SHARDED_BLOCK_EXECUTOR.lock().deref(),
-            partitioned_txns,
-            state_view,
-            maybe_block_gas_limit,
-        )?)
+        if get_remote_sharding() {
+            Ok(V::execute_block_sharded(
+                REMOTE_SHARDED_BLOCK_EXECUTOR.lock().deref(),
+                partitioned_txns,
+                state_view,
+                maybe_block_gas_limit,
+            )?)
+        } else {
+            Ok(V::execute_block_sharded(
+                SHARDED_BLOCK_EXECUTOR.lock().deref(),
+                partitioned_txns,
+                state_view,
+                maybe_block_gas_limit,
+            )?)
+        }
```

Comment:
can we move this if-else to `aptos_vm::execute_block_sharded`? 
And probably the `Lazy`'s closer to `ShardedBlockexecutor`?

Reply:
It may not be best to move it there because
(1) execute_block_sharded is a trait. Today it is not implemented by other executors, but I in future we may want to implement them.
(2) Also the first param is a Generic of type &ShardedBlockExecutor<S, E> where E: ExecutorClient<S>, and that E can be either LocalExecutorClient<S> or RemoteExecutorClient<S>.

I think it is better if executors like AptosVM::execute_block_sharded or PtxBlockExecutor::execute_block_sharded don't know about creating the ShardedBlockExecutor obj.

Reply:
@manudhundi - I think @msmouse 's point is moving the logic of creating a sharded block executor to the `ShardedBlockexecutor` module - we can do it as an helper function (let's say `create_shared_block_executor`). Currently it is scattered in many places which is not very clean.

Reply:
I am fine with creating a helper function, but it will not do much. The Lazy business is specifically for the benchmark logic. So it will just another helper function in the same file.

Reply:
As discussed moved some of the ShardedBlockExecutor create logic out of this file

Reply:
Let's say we keep the if-else here. Moving all the `Lazy`s and relevant functions over to `ShardedBlockExecutor`'s module should be doable? 
We should be able to `use` only `REMOTE_SHARDED_BLOCK_EXECUTOR` and `SHARDED_BLOCK_EXECUTOR` here relatively easily?

I might also suggest remove most of the generics and just use `Arc<dyn>`s but that might sound too jerky here, so consider not having seen this. :trollface:

Reply:
Ok moved all the Lazy business out of chunk_output.rs

Will take the liberty to skip the Arc<dyn> because
(i) It involves little larger scope of changing the trait VMExecutor::execute_block_sharded (and associated implementations)
(ii) I somewhat believe we should use Generics instead of Dyn wherever we can because it is faster.
Hope it is ok